### ISSUE TITLE
Graph not showing in Temperature controller screen when binding to a iSpindel 

### ISSUE THREAD

**Issue by simonboydfoley** [thorrak/fermentrack#490]
I keep having this issue where if I have an iSpindel gravity sensor created in fermentrack (I have 5 defined) and it is bound to a temperature controller , the graph shows up in the gravity sensor but not the temperature controller.  I  keep having to stop brewing the beer (end brewing) and then restarting again before I can get the graph to show in the temperature controller dashboard.

**Comment by Garzlok** [comment]
I don’t have an iSpindel, but I can tell you that it happens to me. If I start a graph, for certain in Fridge Constant to Beer Constant, a temp controller graph does not start (no matter how much time iI give it). The only consistent way I’ve found to get the graph started is I must reboot my pi.

**Comment by simonboydfoley** [comment]
@Garzlok thanks mate ... Just rebooted my pi and it started to work.

This is a really annoying bug as it ruins your fermentaton profile. I have given up on the iSpindel because of this bug as fermentrack is just too unreliable and screws up your SG at a vital time of the fermentaton process. I have gone back to manual Gravity monitoring and calculating by hand.

Fermentrack also does not allow you to edit the raw data to correct the missing data points at the beginning of the session. So you can not even correct for this bug.

It renders fermentrack useless unless you know of your reboot workaround. 

Really appreciate your post

**Comment by Garzlok** [comment]
No problem on the reboot tip.

I won’t go as far and say that it renders Fermentrack useless, since the primary objective of Fermentrack is to control fermentation temperatures, which it will still do. As far as Gravity, you still get a reading that can show the gravity trend, it’s just not graphed on the controller dashboard.

**Comment by lbussy** [comment]
Well, technically Fermentrack is the front-end and a BrewPi-derivative controller is doing the temp control.  To that end, if the UI does not work, Fermentrack does not work.

Have you checked the JAva console when you have the issues to see what's happening?

**Comment by Garzlok** [comment]
I hear and understand what you're saying, but you're getting a little too deep in the weeds for me as far as "what" controls "what". All I'm saying is, yes one part of Fermentrack has picked-up a bug and not doing what it's supposed to do, but Fermentrack is still picking up and displaying gravity readings, so as a homebrewer, you still have a general understanding of your yeast attenuation....and your part of Fermentrack is still reliably chugging away.

As far as checking the Java console....I would need a little help with how to do this, but I'll be more than happy to better document what I see happening.

**Comment by simonboydfoley** [comment]
For me a specific gravity reading on its own is not what I use fermentrack for. I'm looking for the decrease in rate of the fall in Gravity to know the end of peak fermentation and the start of maturity phase when I need to lift the temperature to start clearing down the yeast artifacts. So useless is a bit harsh on my part but missing the starting gravity and that fermentation turning point to raise the temp is a real pain in the arse.

I'll see if I can see any tomcat errors after I have finished brewing these 5 batches of beer ☺️

**Comment by lbussy** [comment]
> As far as checking the Java console....I would need a little help with how to do this, but I'll be more than happy to better document what I see happening.

Generally: Shift + Ctrl + K will open the web console.  If not, let us know what browser.

**Comment by simonboydfoley** [comment]
By the way if it helps ..... O have packet sniffed the  network traffic and I can confirm the iSpindel and the temperature controller is sending updates to fermentrack on the wire. This is totally a fermentrack by issues

**Comment by thorrak** [comment]
Huh. OK. This sounds like a caching problem.

**Comment by thorrak** [comment]
I think I found the bug. When you have a moment, update to the latest dev branch and let me know if the bug is resolved. If this didn't resolve it, I have another fix - but it requires more refactoring so I'd rather not go that route if this takes care of the problem.

**Comment by Garzlok** [comment]
I pulled the change to my Pi and I can confirm a graph started on Fridge Constant, Beer Constant, and Beer Profile with a simple refresh of the browser while on my controller dashboard page.

**Comment by thorrak** [comment]
> I pulled the change to my Pi and I can confirm a graph started on Fridge Constant, Beer Constant, and Beer Profile with a simple refresh of the browser while on my controller dashboard page.

Soooooo fixed? 🤞🏻

**Comment by simonboydfoley** [comment]
I won't be able to tell without ruining my 5 brew batches / BrewPi chart. So that would be 2 weeks  from now I can test. I maybe able to build a second BrewPi and 6th iSpindel and test before then ... Let me see what parts I have when I get home.

**Comment by thorrak** [comment]
> I won't be able to tell without ruining my 5 brew batches / BrewPi chart. So that would be 2 weeks from now I can test. I maybe able to build a second BrewPi and 6th iSpindel and test before then ... Let me see what parts I have when I get home.

No rush. There is no drawback to the fix I pushed live, so even if it is still broken there wouldn't be a change to revert the code - I'd just need to find another fix.

**Comment by Garzlok** [comment]
> Soooooo fixed? 🤞🏻

Yeah, I think so. 

I know (for the last couple months) anytime I started a graph in either a Beer or Fridge Constant, I had to reboot my Pi to get the graph to start. Last night, after implementing your change, graphing occurred at the designated log intervals and a simple browser refresh displayed them.

**Comment by lbussy** [comment]
Seems like restarting Fermentrack would have the same effect.

**Comment by Garzlok** [comment]
Yeah, but a Pi reboot is 1 command line, restarting Circus is 4 command lines. 🤣

**Comment by lbussy** [comment]
> Yeah, but a Pi reboot is 1 command line, restarting Circus is 4 command lines. 🤣

... or one click for the developer to put in.  You know, if he did anything ... ;)

**Comment by uSlackr** [comment]
This leads to the question of whether we could stop/pause and resume a fermentation.

**Comment by thorrak** [comment]
> This leads to the question of whether we could stop/pause and resume a fermentation.

As in, the log? Yes.

As in, the fermentation itself? I'm not a yeast-whisperer. :)

**Comment by uSlackr** [comment]
Damn 😊

Yes, the log.


\\Greg

**Comment by thorrak** [comment]
> Damn 😊 Yes, the log. \\Greg

There should be a button to pause logging. If you're asking about stopping/restarting, that doesn't exist yet, but if it's really something you want I'd recommend opening an issue for it here. It shouldn't be difficult, but would require a bit of development.

**Comment by uSlackr** [comment]
Never saw the Pause button.  But t here it is!

\\Greg

**Comment by modul8** [comment]
Updated to this version this morning on a rpi3b+ running buster. It now won't run. After triggering update and trying to refresh screen after about 10 mins i get a 502 bad gateway error. Buster is still running and i can ssh in. Any idea where to look? Cheers
